### DISCUSSION TITLE
mz_stream_mem_set_size performance issue

### DISCUSSION THREAD

**Discussion by zhuda** [nmoinvaz/minizip#614] [Q&A]
Hi, I notice the mz_stream_mem_set_size performance dramatically slow down when writing a lot entries. In my test code I wrote out more than 650000 entries with `mz_zip_writer_close_entry` and it finally called this function. 
After profiling my understanding is the buffer size is continuously out of usage, so the `mz_stream_mem_set_size` continuouslly allocate new buffer, copy old buffer values, deallocate the old buffer. 
So question: Is there any API can control the memory management mechanism? Like allocate a larger buffer at very beginning? or allocate more spaces in each iterator? 
![image](https://user-images.githubusercontent.com/3140101/156692136-244ce2bc-5ee2-47ee-afd8-4837532135ae.png)

**Comment by nmoinvaz** [comment]
There is only `mz_stream_mem_set_grow_size` which can control the increment of which the memory stream can grow. Also if you have a general solution that would benefit everybody I would be open to PRs.
